File: MotorControle/Platform_IO_Board_Motor_Controle/test/testComm/test_ControlMessage.cpp
#include "test_ControlMessage.h"

void testControlMessage::isNormalInisialisationOk()
{
    int data[] = {1,2,3};
    int size = 3;
    int type = 1;
    
    ControlMessage msg(type,size,data);

    TEST_ASSERT_EQUAL(1,msg.getType());
    TEST_ASSERT_EQUAL(3,msg.getPayLoadSize());
    
    for(int x=0; x < size;x++)
    {
        TEST_ASSERT_TRUE(data[x] == msg.getPayload()[x]);
    }
}

void testControlMessage::isTypeInisialisationOk()
{
    int type = 1;
    
    ControlMessage msg(type);

    TEST_ASSERT_EQUAL(1,msg.getType());
    TEST_ASSERT_EQUAL(0,msg.getPayLoadSize());
    TEST_ASSERT_EQUAL(0,msg.getPayload());
}

void testControlMessage::isBadInisalistionWorking()
{
    int type = 1;
    int payloadSize = 2;

    ControlMessage msg(type,payloadSize);

    TEST_ASSERT_EQUAL(1,msg.getType());
    TEST_ASSERT_EQUAL(0,msg.getPayLoadSize());
    TEST_ASSERT_EQUAL(0,msg.getPayload());
}


void testControlMessage::test()
{
        RUN_TEST(isNormalInisialisationOk);
        RUN_TEST(isTypeInisialisationOk);
        RUN_TEST(isBadInisalistionWorking);
}

File: Application/test/TestComm/testControlMessage.py
import unittest
from Comm.ControlMessage import ControlMessage

class TestControlMessage(unittest.TestCase):
    def test_init(self):
        data = [1,2,3,4]
        msg = ControlMessage(1,[1,2,3,4])

        self.assertEqual(1,msg.getType())
        self.assertEqual(4,msg.getPayloadSize())

        for x in range(0, len(data)):
            self.assertEqual(data[x], msg.getPayload()[x])

    def test_OnlyTypeInit(self):
        msg = ControlMessage(1)

        self.assertEqual(1, msg.getType())
        self.assertEqual(0, msg.getPayloadSize())
